src/providers/tpm_provider/key_management.rs

```diff
@@ -208,18 +200,12 @@ impl TpmProvider {
                 Err(utils::to_response_status(e))
             })?;
 
-        let key = RsaPublicKey {
-            // To produce a valid ASN.1 RSAPublicKey structure, 0x00 is put in front of the positive
-            // modulus if highest significant bit is one, to differentiate it from a negative number.
-            modulus: IntegerAsn1::from_unsigned_bytes_be(pub_key_data),
-            public_exponent: IntegerAsn1::from_signed_bytes_be(PUBLIC_EXPONENT.to_vec()),
-        };
-        let key_data = picky_asn1_der::to_vec(&key).or_else(|err| {
-            error!("Could not serialise key elements: {}.", err);
-            Err(ResponseStatus::PsaErrorCommunicationFailure)
-        })?;
-
-        Ok(psa_export_public_key::Result { data: key_data })
+        Ok(psa_export_public_key::Result {
+            data: utils::pub_key_to_bytes(
+                pub_key_data,
+                usize::try_from(key_attributes.key_bits / 8).unwrap(), // should not fail on supported platforms
```

Comment:
Not at all in scope with this PR, but maybe we should implement a `From` the error returned by `TryFrom` to `ResponseStatus` to be able to remove those `unwrap`, even though they should not happen on the correct platforms.

Reply:
That would be from this type: https://doc.rust-lang.org/std/num/struct.TryFromIntError.html

Reply:
I'm guessing that would be a `GenericError`? Or would it be some sort of hardware error?

Reply:
`GenericError` with a `error!` log message saying why seems fine I think